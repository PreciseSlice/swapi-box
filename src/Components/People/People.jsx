import React from 'react';

const People = (props) => {
  return (
    <div>
      <h3>I AM A PERSON</h3>
    </div>
  );
};

export default People;