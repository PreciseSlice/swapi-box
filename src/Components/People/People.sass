.people 
  color: #feda4a