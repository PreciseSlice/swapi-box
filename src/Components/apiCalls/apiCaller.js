export const callFetch = async url => {
  const getData = await fetch(url);
  const jsonData = await getData.json();

  return await jsonData;
};

export const getFilms = async url => {
  try {
    const fetchFilmData = await callFetch(url);

    const title = await fetchFilmData.title;
    const episodeId = await fetchFilmData.episode_id;
    const openingCrawl = await fetchFilmData.opening_crawl;
    const releaseDate = await fetchFilmData.release_date;

    return await { title, episodeId, openingCrawl, releaseDate };
  } catch (er) {
    const error = new Error('getFilms failed to fetch filmData');
    return error;
  }
};

export const getPeople = async url => {
  try {
    const fetchPeopleData = await callFetch(url);

    const peopleMap = fetchPeopleData.results.map(async person => {
      const name = person.name;

      const fetchHomeworldData = await callFetch(person.homeworld);
      const homeworld = fetchHomeworldData.name;
      const population = fetchHomeworldData.population;

      const fetchSpeciesData = await callFetch(person.species);
      const speciesName = fetchSpeciesData.name;

      return { name, homeworld, population, speciesName };
    });
    return Promise.all(peopleMap);
  } catch (er) {
    const error = new Error('getPeople failed to fetch peopleData');
    return error;
  }
};

export const getPlanets = async url => {
  try {
    const fetchPlanetData = await callFetch(url);

    const planetMap = fetchPlanetData.results.map(async planet => {
      const name = planet.name;
      const terrain = planet.terrain;
      const population = planet.population;
      const climate = planet.climate;

      const residentMap = planet.residents.map(async resident => {
        const fetchResidentData = await callFetch(resident);

        return fetchResidentData.name;
      });

      const mapReturn = await Promise.all(residentMap);
      const residents = mapReturn.join(', ');

      return { name, terrain, population, climate, residents };
    });
    return Promise.all(planetMap);
  } catch (er) {
    const error = new Error('getPlaenets failed to fetch plaenetData');
    return error;
  }
};

export const getVehicles = async url => {
  try {
    const fetchVehicleData = await callFetch(url);

    const vehicleMap = fetchVehicleData.results.map(async vehicle => {
      const name = vehicle.name;
      const model = vehicle.model;
      const vehicleClass = vehicle.vehicle_class;
      const passengers = vehicle.passengers;

      return { name, model, vehicleClass, passengers };
    });
    return Promise.all(vehicleMap);
  } catch (er) {
    const error = new Error('getVehicles failed to fetch vehicleData');
    return error;
  }
};
