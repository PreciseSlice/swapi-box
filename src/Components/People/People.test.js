/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import People from './People';
import { shallow, mount, render } from 'enzyme';
import { peopleData } from '../apiCalls/apiMockData';

describe('People', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <People
        setPeople={jest.fn()}
        peopleData={peopleData}
        clickHandler={jest.fn()}
        favorites={peopleData}
      />
    );
  });

  it('Exist and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should get people data if it does not have it', () => {

  });

  it('renders a card for each object in the array it is passed', () => {
    expect(wrapper.find('.card').length).toEqual(2);
  });
  

});
