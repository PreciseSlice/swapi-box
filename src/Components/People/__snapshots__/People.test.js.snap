// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Exist and matches snapshot 1`] = `
<People
  clickHandler={[Function]}
  favorites={
    Array [
      Object {
        "homeworld": "planet",
        "name": "Luke Skywalker",
        "population": 100,
        "speciesName": "human",
      },
      Object {
        "homeworld": "planet",
        "name": "C-3PO",
        "population": 12,
        "speciesName": "droid",
      },
    ]
  }
  peopleData={
    Array [
      Object {
        "homeworld": "planet",
        "name": "Luke Skywalker",
        "population": 100,
        "speciesName": "human",
      },
      Object {
        "homeworld": "planet",
        "name": "C-3PO",
        "population": 12,
        "speciesName": "droid",
      },
    ]
  }
  setPeople={[Function]}
>
  <div
    className="peopleContainer"
  >
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "homeworld": "planet",
          "name": "Luke Skywalker",
          "population": 100,
          "speciesName": "human",
        }
      }
      highlighted="highlighted"
      key="Luke Skywalker"
    >
      <div
        className="card highlighted"
      >
        <h1>
          Luke Skywalker
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            homeworld: planet
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 100
          </li>
          <li
            key="function now() { [native code] }2"
          >
            speciesName: human
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          Favorite
        </button>
      </div>
    </Card>
    <Card
      clickHandler={[Function]}
      data={
        Object {
          "homeworld": "planet",
          "name": "C-3PO",
          "population": 12,
          "speciesName": "droid",
        }
      }
      highlighted="highlighted"
      key="C-3PO"
    >
      <div
        className="card highlighted"
      >
        <h1>
          C-3PO
        </h1>
        <ul>
          <li
            key="function now() { [native code] }0"
          >
            homeworld: planet
          </li>
          <li
            key="function now() { [native code] }1"
          >
            population: 12
          </li>
          <li
            key="function now() { [native code] }2"
          >
            speciesName: droid
          </li>
        </ul>
        <button
          onClick={[Function]}
        >
          Favorite
        </button>
      </div>
    </Card>
  </div>
</People>
`;
